include/spdlog/common.h

```diff
@@ -116,16 +116,27 @@ using wstring_view_t = fmt::basic_string_view<wchar_t>;
 using memory_buf_t = fmt::basic_memory_buffer<char, 250>;
 using wmemory_buf_t = fmt::basic_memory_buffer<wchar_t, 250>;
 
+template<class T>
+using remove_cvref_t = typename std::remove_cv<typename std::remove_reference<T>::type>::type;
+
+// clang doesn't like SFINAE disabled constructor in std::is_convertible<> so have to repeat the condition from basic_format_string here,
+// in addition, fmt::basic_runtime<Char> is only convertible to basic_format_string<Char> but not basic_string_view<Char>
+template<class T, class Char = char>
+struct is_convertible_to_basic_format_string
+    : std::integral_constant<bool,
+          std::is_convertible<T, fmt::basic_string_view<Char>>::value || std::is_same<remove_cvref_t<T>, fmt::basic_runtime<Char>>::value>
+{};
+
 #ifdef SPDLOG_WCHAR_TO_UTF8_SUPPORT
 #    ifndef _WIN32
 #        error SPDLOG_WCHAR_TO_UTF8_SUPPORT only supported on windows
 #    else
 template<typename T>
 struct is_convertible_to_wstring_view : std::is_convertible<T, wstring_view_t>
 {};
+
 template<class T>
-struct is_convertible_to_wformat_string : std::is_convertible<T, fmt::wformat_string<>>
-{};
+using is_convertible_to_wformat_string = is_convertible_to_basic_format_string<T, wchar_t>;
```

Comment:
Does it compile if SPDLOG_WCHAR_TO_UTF8_SUPPORT (and hence is_convertible_to_wformat_string ) are not defined ?

Anyway I think it is cleaner just to remove the ```is_convertible_to_wformat_string``` definitions completely (lines 134-147) and do something like:

```c++
template<class T>
struct is_convertible_to_any_format_string
    : std::integral_constant<bool, is_convertible_to_basic_format_string<T, char>::value || is_convertible_to_basic_format_string<T, wchar>::value>
{};
```

Reply:
`is_convertible_to_wformat_string` is defined the same way as `is_convertible_to_wstring_view` but a quick search in the repo showed no uses of either outside of `logger::log` SFINAE check if `msg` is convertible to a string view or format string. I used `is_convertible_to_wstring_view` as an example for `is_convertible_to_wformat_string` since they are both very similar. Might be cleaner to remove both in this case.

Reply:
yes. please remove them